Add support for theme icons

awinny opened the pull request:
**Describe pull-request**
- Updated icon example page to support new icon setup
- Fix for bug in theme component to handle dynamic change of theme in a correct way
- Changed initiation of default icons as well as theme icon handling
- Added icon part to the store
</br>

**Solving issue**
Fixes: #144 
</br>

**Additional context**
This theature is connected to this task:
https://github.com/scania/scania-theme/pull/7

awinny committed:
Add support for theme icons
- Updated icon example page to support new icon setup
- Fix for bug in theme component to handle dynamic change of theme in a correct way
- Changed initiation of default icons as well as theme icon handling
- Added icon part to the store

helloimela commented on public/templates/icons-page.js:
```diff
@@ -33,20 +99,7 @@ export default {
     </div>
 
     <h2>Solid icons</h2>
```
Change "solid icons" to something general

helloimela reviewed (changes requested):
Cool!
Please check icons-page works in IE.
And also change the section title in icon-page.

helloimela commented:
Future improvements:
- Refactor store so we can load icons on runtime

awinny committed:
Browser compability fixes
- Added missing ie support for custom elements
- Simplified icon custom element to solve issue in ie11

helloimela reviewed (approved):
Tested & looks good!
